net-square-designs/Errnd
# authenticate users

#### What does this PR do?
It authenticates a user
#### Description of Task to be completed
- create sign up functionality
- create login functionality
- create database as well as users tables
- validate all input as well as take care of edge cases
#### How should this be manually tested?
* Clone the repo
* Execute the following commands on your terminal

> * git checkout ` ft-sign-up-a-user`.
> * npm install
> * npm run start:dev
> * using postman navigate to **POST** `http://localhost:3005/api/v1/auth/signup`.
> * add this json in the raw filed `{
"role": "customer",
"email": "newcustomer@errnd.com",
"password": "passworD45@@",
"username": "newcustomer"
}`
> * click send
> * You should see a response body containing the message `User created successfully`
> * using postman navigate to **POST** `http://localhost:3005/api/v1/auth/login`.
> * add this json in the raw filed `{
"email": "newcustomer@errnd.com",
"password": "passworD45@@"
}`
> * click send
> * You should see a response body containing the message `Welcome newcustomer`
#### Any background context you want to provide?
Previously, there was no way to authenticate a user

#### Screenshots (if appropriate)
N/A
